# Problem 9.4.1 (Rental Price Index of Zurich, cf. [33], Reg3, Problem 1).

The data set RPI-ZH.dat contains the Rental Price Index of the city of Zurich (RPI), the Mortgage Interest of the Zu ̈rcher Kantonalbank (MI) and the Consumer Price Index of the canton of Zurich (CPI) for each quarter between 1994 and May 2005.
From legal regulations on rental rates, it is clear that the mortgage rate and the consumer price index strongly affect the rental price.

Consider the following model

RPI=β0 +β1 ·CPI+β2 ·MI+ε.

Assume that the errors ε are independent normally distributed with expected value 0 and variance σ2.

In [1]:
import os
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import statsmodels.formula.api as smf
import scipy as scp
import warnings
warnings.filterwarnings('ignore')

In [5]:
df = pd.read_csv('./04 Datasets/RPI-ZH.dat', sep='\t')
print(df.head())

  Month  Year    RPI   MI   CPI
0   Feb    94  98.50  6.0  95.1
1   Mai    94  96.42  5.5  94.6
2   Aug    94  96.55  5.5  95.1
3   Nov    94  96.80  5.5  95.1
4   Feb    95  97.16  5.5  96.6


### a) Fit the model to the data. Report the estimated coefficients. Overall, do the explanatory variables influence the target values?

In [7]:
model = smf.ols('RPI ~ CPI + MI', data=df).fit()
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                    RPI   R-squared:                       0.927
Model:                            OLS   Adj. R-squared:                  0.924
Method:                 Least Squares   F-statistic:                     272.7
Date:                Mon, 22 Apr 2024   Prob (F-statistic):           3.74e-25
Time:                        20:38:58   Log-Likelihood:                -64.075
No. Observations:                  46   AIC:                             134.2
Df Residuals:                      43   BIC:                             139.6
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept    -68.7083     10.226     -6.719      0.0

estimate of sigma

In [9]:
np.sqrt(model.scale)

1.0077560277821491

REMARKS:    We find $\hat{\beta}_0=-68.7083$, $\hat{\beta}_1=1.6183$, $\hat{\beta}_2=2.0069$ and $\hat{\sigma}=1.00776$.
            The overall test has $F=272.7$ on $(2,43)$ degrees of fredom and its P-value is much smaller than $5\%$.
            Therefore at least one of the coefficients $\beta_1$ and $\beta_2$ is different of zero.

### (b) R-squared

In multiple regression  the coefficient of determination, often called $R^2$ is identical to the squared correlation between the response variable and the fitted values. The coefficient of determination is a measure of the linear relationship between the response variable and the fit.

In [10]:
model.rsquared

0.9269148955002268